README.md

```diff
@@ -1,28 +1,76 @@
+
 # Hardware Information API (hwapi)
 
 The repo contains the API server and client for retrieving hardware information.
 
-## Prerequisites for Deploying Locally
+## Prerequisites for deploying locally
 
 - Install [docker](https://docs.docker.com/engine/install/ubuntu/) and
   [setup permissions](https://docs.docker.com/engine/install/linux-postinstall/)
   for server deployment
 - Install `rust` and `cargo` for client deployment
   (https://www.rust-lang.org/tools/install)
 
-## Running API Server With Docker
+Also, to run the server locally, make sure that another application doesn't use the port `8080`.
+
+## Running API development server instance with Docker
+
+Go to the `server/` directory in the project. Since the server needs the data from the DB to work with,
+there are several options regarding how you can stand up the environment.
+
+### Using pre-populated database
+
+You can use this option if you already have the SQLite DB file and want to use it. Make sure that the DB
+is located under the `./server/` directory.
+
+You can stand up the environment by running the following commands:
+
+```bash
+# Assuming that the path to the DB file is ./hwapi.db
+docker-compose build --build-arg IMPORT_TOOL_PATH="" --build-arg DB_URL=sqlite:///./hwapi.db hwapi-dev
```

Comment:
Hmm, does the Dockerfile image even need the `IMPORT_TOOL_PATH` arg at all if the import tool can just be run outside on the host / in a VM or container started up by a CI system?

Reply:
I assumed importing data is the default behaviour, so you can run `docker-compose up` and get the container with the populated data. I thought that this is what we also want to have by default in production, that’s why I’ve set the default value, not an empty string
Do you think the default behaviour should be not to populate the DB at all?

Reply:
If the Dockerfile can be kept simple, I'd keep it simple. 1) You have the import tool and 2) you have the Dockerfile that can use the DB, no matter how sourced. This is composable, and I don't think you need to specifically need to provide an option for the image build to execute that script.

Reply:
For `docker-compose up` to work for a fresh checkout, IMO best option would be a small example data with some toy data.